create_bags/bag_creator.py

```diff
@@ -1,18 +1,67 @@
-import logging
-
+import json
 from configparser import ConfigParser
+from subprocess import PIPE, Popen
+
 from .archivesspace import ArchivesSpaceClient
+from .helpers import create_tag, format_aspace_date, get_dates
 
 
-class BagCreator():
-    """docstring for BagCreator"""
+class BagCreator:
 
     def __init__(self):
-        logging.basicConfig(
-            datefmt='%m/%d/%Y %I:%M:%S %p',
-            filename='iiif_preparer.log',
-            format='%(asctime)s %(message)s',
-            level=logging.INFO)
         self.config = ConfigParser()
         self.config.read("local_settings.cfg")
-        self.ignore_list = self.config.get("IgnoreList", "ignore_list")
+        self.dest_location = self.config.get("Locations", "dest_location")
+        self.dart_command = self.config.get("DART", "dart")
+
+    def run(self, refid, rights_ids, files):
+        """
+        Args:
+        refid (str)
+        rights_ids (array)
+        """
+        # directory_to_bag = "some directory"
+        self.as_client = ArchivesSpaceClient(baseurl=self.config.get(
+            "ArchivesSpace", "baseurl"), username=self.config.get(
+            "ArchivesSpace", "username"), password=self.config.get(
+            "ArchivesSpace", "password"))
+        self.refid = refid
+        self.ao_uri = self.as_client.get_ao_uri(self.refid)
+        ao_data = self.as_client.get_ao_data(self.ao_uri)
+        dates = format_aspace_date(get_dates(ao_data))
+        self.job_params = self.construct_job_params(rights_ids, files, dates)
+        self.create_dart_job()
+        return self.refid
+
+    def construct_job_params(self, rights_ids, files, dates):
+        """Formats information for DART job parameters
+
+        Args:
+            rights_ids (array): list of rights ids
+            files (array): list of full filepaths
+            dates (tuple): begin and end dates
+
+        Returns a dictionary"""
+        job_params = {"workflowName": "Digitization Workflow"}
+        job_params['packageName'] = "{}.tar".format(self.refid)
+        job_params['files'] = files
+        tags = []
+        tags.append(create_tag("ArchivesSpace-URI", self.ao_uri))
+        tags.append(create_tag("Start-Date", dates[0]))
+        tags.append(create_tag("End-Date", dates[1]))
+        tags.append(create_tag("Origin", "digitization"))
```

Comment:
I wonder if you can do this all in a static dictionary - it might make the overall shape of that data a little more obvious.